source/CardGames.py

```diff
@@ -291,17 +284,30 @@ def showWinner(players):
     winner = handTotals[max(handTotals.keys())] 
     return winner
 
-wins = {}
-
-def updateLeaderboard(winner):
-    global wins
-    if winner not in wins:
-      wins[winner] = 1
-    else:
-      wins[winner] += 1
-    
-def showLeaderboard(wins):
-    leaderboard = "\nLEADERBOARD:\n"
-    for winner in wins:
-      leaderboard += f"{winner.name}: {wins[winner]} wins\n"
-    print(leaderboard)
+source_folder = os.path.join(os.getcwd(), '..', 'source')
+leaderboard_file = os.path.join(source_folder, 'leaderboard.txt')
+
```

Comment:
This function looks really good, really clean and not overly complicated. One thing to check, when you go to write in a new player to the leaderboard in line 305, it looks like the "1" added might be a string. This would cause an error if this player was to win again and you were to try to add "+1" to this value in line 298. Make sure the "1" getting added initially is a string.

Reply:
I see why you'd think that but I tested it out with adding an extra player. After their name is incorporated in the leaderboard, it updates accordingly.